### ISSUE TITLE
use_state, method not found in `&mut Ui`

### ISSUE THREAD

**Issue by aspcartman** [ryo33/egui_hooks#16]
Seems that the use of the UseHookExt doesn't work anymore.
egui_hooks = "0.1.2"
egui = "0.27.0"
eframe = "0.27.0"

```
use egui_hooks::UseHookExt;
...
let text = &ui.use_state(|| String::new(), ());
```

![Screenshot 2024-05-15 at 13 05 12](https://github.com/ryo33/egui_hooks/assets/844194/ca8bcd69-36d0-4182-8cb9-e8837b18ac48)

```
let &text = &UseHookExt::use_state(ui, || String::new(), ())
                .into_var().deref_mut();
```

![Screenshot 2024-05-15 at 13 07 13](https://github.com/ryo33/egui_hooks/assets/844194/ab68c58a-c48d-4b9d-80e0-b070b6352dca)

Seems weird, yet I'm new to Rust : )

Am I doing anything wrong?

![Screenshot 2024-05-15 at 13 09 27](https://github.com/ryo33/egui_hooks/assets/844194/f319deb9-5c87-46b0-ab90-df9320add2ae)

**Comment by aspcartman** [comment]
Fixes by putting * in the cargo.toml of egui_hooks. Otherwise have to downgrade the project's egui down to 0.24.

https://github.com/aspcartman/egui_hooks/blob/main/Cargo.toml

**Comment by ryo33** [comment]
Hi, thanks for reporting.

I've released `egui_hooks@0.4.0` now, which should work with `egui@0.27`. Can you try it?
